<!DOCTYPE html>
<html lang="en" xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Alaska Flora : Blog 5: Phenotype data entry and sharing</title>
    <meta name="viewport" content="width=device-width, initial-scale=1"/>
    <link href="https://fonts.googleapis.com/css?family=Montserrat:400,300,600" rel="stylesheet" type="text/css"/>
    <link rel="stylesheet" href="../css/normalize.css"/>
    <link rel="stylesheet" href="../css/skeleton.css"/>
    <link rel="stylesheet" href="../css/override.css"/>
    <link rel="icon" type="image/png" href="../img/favicon.png"/>
    <link rel="alternate" type="application/rss+xml" title="Alaksa Flora Development Blog" href="blogrss.xml" />
  </head>
  <body>
    <div class="container">
      <div class="row">
        <div class="nine columns" style="margin-top: 5%">
          <h1 id="blog-4-phenotype-data-entry-and-sharing">Blog 4: Phenotype data entry and sharing</h1>
          <p><em>Posted by <a href="people.html#cam">Cam</a> on 2019-02-28</em></p>
          <p>I got diverted from “name wrangling” these past few weeks, and have been thinking about morphological characters and pehenotypes. Among the goals of the NSF-funded project is to develop a taxon × character matrix for all the ca. 2,300 taxa we will eventually have in the Flora. This matrix can be used for analyses and in descriptions of taxa, but the main use will be in an interactive identification key for Alaskan plants. An interactive key differs from a traditional dischotomous key primarily in allowing the user to start with any character they may have handy, rather than pre-determining the first character couplet. An interactive key is essentially the cumulative application of character filters on the taxon × character matrix until a single or small number of taxa remain.</p>
          <p>Our plan in the proposal was to assemble data on enough characters to reduce the number of taxa to a size where visual scanning of thumbnail images would permit rapid selection of a final identified taxon. The primary trade-off for us in chosing the numnber of characters to score is “descriminatory capability” (i.e., reducing the potential taxa list as much as possible) vs. time needed to assemble the matrix. Since we intend the interactive key to be usable by interested but relatively untrained members of the public, the set of available characters is fairly constrained to those that do not need botanical experience (e.g., not “placentation type”) or special equipment (e.g., not “hairs stellate”).</p>
          <p>After discussion among the Flora of Alaska Executive Committee, and some initial trial and error, we have settled on these 12 characters:</p>
          <ul>
          <li>Whole plant phenotypes
          <ul>
          <li>Plant habit: tree, shrub, grass-like or forb</li>
          <li>Plant stem: caulescent or acaulescent</li>
          </ul></li>
          <li>Leaf phenotypes
          <ul>
          <li>Leaf insertion: alternate or opposite</li>
          <li>Leaf form: simple or not (including palmate, pinnate, trifoliate)</li>
          <li>Leaf margin: entire or not</li>
          <li>Leaf primary venation: parallel or not</li>
          </ul></li>
          <li>Flower/fruit phenotypes
          <ul>
          <li>Inflorescence of multiple flowers: yes or no (solitary flowers)</li>
          <li>Flower with pedicel: “stalked” or not</li>
          <li>Flower symmetry: actinomorphic or zygomorphic<br />
          </li>
          <li>Flower color: blue, brown, green, orange, pink, purple, red, white, or yellow</li>
          <li>Ovary position: inferior, semi-inferior or superior</li>
          <li>Fruit fleshiness: dry or fleshy</li>
          </ul></li>
          </ul>
          <h2 id="phenotype-definition-interoperability">Phenotype definition interoperability</h2>
          <p>As you know by now, <em>interoperability</em> is the name of the game with this project, and as we set out to record these morphological data, we need to try to maximize the possibility that they can be re-used by other projects. So, “how best to standardize and share phenotype data?” Vast numbers of taxon × character matrices (and individual × character matrices) have been assembled (see e.g., <a href="https://morphobank.org/">Morphobank</a>, but most of the matrices cannot easily be “aligned” with other matrices: even though the taxa may overlap, the characters are seldom standardized and often have definitions that are ideosyncratic to a particular project. This situtation is in contrast to the taxa × sequence data, which <em>can</em> generally be aligned, and thus can be contributed to vastly resources like <a href="https://www.ncbi.nlm.nih.gov/genbank/">Genbank</a>.</p>
          <p>To answer the above questions about character standardization, we need to take a detour into the nuts and bolts of systematically recording phenotype data, whether for identification or for mophological analysis purposes. The simplest “model” of phenotype is <code>Character + Character state</code>, e.g., “leaf insertion into stem” (= <code>Character</code>) + “opposite” (= <code>Character state</code>). This was the approach taken by the the pioneering <a href="https://www.delta-intkey.com/">DELTA</a> system (DEscription Language for TAxonomy; see also <a href="http://downloads.ala.org.au/p/Open%20DELTA">OpenDelta</a> and <a href="http://freedelta.sourceforge.net/fde/">FreeDelta</a>), as well as that of MacClade, and its descendant <a href="https://www.mesquiteproject.org/">Mesquite</a>. This model was also adopted by the <a href="https://github.com/tdwg/sdd">Structured Descriptive Data</a> standard of <a href="https://www.tdwg.org/">TDWG</a>.</p>
          <p>This two-part descriptive approach was critiqued in an <a href="http://dx.doi.org/10.2307/25065431">important paper</a> by Pullan et al. (2005), who had proposed a three-part “descriptive element” (part of the Prometheus II project): <code>Structure + Property + Score</code>. And in another lineage of this phenotype conversation, growing out of the gene annotation and Open Biological and Biomedical Ontology (OBO) community, <a href="https://www.usd.edu/faculty-and-staff/Paula-Mabee">Paula Mabee</a>, <a href="https://www.mendeley.com/profiles/jim-balhoff/">Jim Balhoff</a> and <a href="https://phenoscape.org/">Phenoscape</a> collegues have developed the <a href="https://wiki.phenoscape.org/wiki/EQ_for_character_matrices">Entity-Quality formalism</a>, where a complex <code>Quality</code> “inheres” in an <code>Entity</code>. The three approaches line up so:</p>
          <table class="blog5_tab">
          <tr>
          <th>
          System
          </th>
          <th>
          Thing
          </th>
          <th>
          Quality
          </th>
          <th>
          Quality value
          </th>
          </tr>
          <tr>
          <td>
          DELTA, Mesquite, etc.
          </td>
          <td colspan="2">
          Character<br/><i>(Leaf insertion)</i>
          </td>
          <td>
          State<br/><i>(opposite)</i>
          </td>
          </tr>
          <tr>
          <td>
          Prometheus II<br/>(Pullen et al. 2005)
          </td>
          <td>
          Structure<br/><i>(Leaf)</i>
          </td>
          <td>
          Property<br/><i>(insertion geometry)</i>
          </td>
          <td>
          Score<br/><i>(opposite)</i>
          </td>
          </tr>
          <tr>
          <td>
          EQ Model<br/>(Phenoscape)
          </td>
          <td>
          Entity<br/><i>(Leaf)</i>
          </td>
          <td colspan="2">
          Quality<br/><i>(opposite insertion,<br/>= subclass of ‘insertion geometry’)</i>
          </td>
          </tr>
          </table>
          <p>One major advantage Ontology based…</p>
          <p>Two are needed: a structural ontology… and a</p>
          <p>Functionally, it doesn’t matter how you score the character in a local database…</p>
          <p>https://www.researchgate.net/publication/221502264_A_Universal_Character_Model_and_Ontology_of_Defined_Terms_for_Taxonomic_Description/download</p>
          <p>The authors also</p>
          <p>We just hired a project researcher, Brian</p>
          <p>Taxon that inverse(‘member of taxon’) some (Organism that ‘has phenotype’ some ‘Leaf insertion not alternate’)</p>
          <p>Organism that ‘has phenotype’ some ‘Leaf insertion not alternate’</p>
          <p>Phenotype that ‘Leaf insertion not alternate’</p>
          <p>RDF open world</p>
          <p>A &gt; B :: B rdfs:subClassOf A A &lt; B :: A rdfs:subClassOf B A == B :: A owl:equivalentClass B A &gt;&lt; B :: ( no statement needed, assumed by OWA ) A | B :: A owl:disjointWith B</p>
          <ol type="1">
          <li>The FNA Categorical Glossary: http://www.huntbotanical.org/databases/show.php?4 Can be queried here: http://fmhibd.library.cmu.edu/HIBD-DB/FNA/findrecords.php</li>
          </ol>
          <p>This resource covers a large proportion of the terms used in the FNA, but is not complete.</p>
          <ol start="2" type="1">
          <li>The FloraTerms TDWG resource: https://terms.tdwg.org/wiki/FloraTerms</li>
          </ol>
          <p>This project originated from the development of the FNA Categorical Glossary. It extracts and enhances terms from the FNA Categorical Glossary, through the use of CharaParser on the FNA. It was intended to be a simple, collaborative space for the agreement on definitions by botanists. Its terms are more simple than PATO’s.</p>
          <p>In fact, a few of us proposed that we should begin contributing to this resource again, and attempt to revive it under our new project building a Carex ontology for a treatment authoring tool.</p>
          <ol start="3" type="1">
          <li>Planteome has been amenable to accepting contributed terms, and so another approach would be creating a set of definitions de novo, and contribute what is developed.</li>
          </ol>
          <p>Lastly, a final consensus of our discussion was that we will need an answer to this question for our work in the future. If we can collaborate in some way to advance this cause, please reach out.</p>
        </div>
        <div class="three columns" style="margin-top: 5%">
          <a class="button" style="width:100%" href="project.html">Project Info</a><br/>
          <a class="button" style="width:100%" href="blog.html">Dev Blog</a><br/>
          <a class="button" style="width:100%" href="products.html">Products</a><br/>
          <a class="button" style="width:100%" href="people.html">People</a>
          <a class="button" style="width:100%" href="../index.html">Home</a><br/>
          <img src="../img/menupic.jpg" alt="menu picture" width="100%"/>
        </div>
      </div>
    </div>
  </body>
</html>
